tests/integration/tier0/check-user-response/test_user_response.py

```diff
@@ -16,13 +16,26 @@ def test_check_user_response_user_and_password(convert2rhel):
         # "Password" and raises the assertion error.
         assert c2r.expect_exact(["Username", "Password"], timeout=300) == 0
         # Provide username, expect password prompt
-        c2r.sendline(env.str("RHSM_USERNAME"))
-        c2r.expect_exact("Password: ")
-        c2r.sendline()
-        assert c2r.expect_exact(["Password", "Enter number of the chosen subscription"], timeout=300) == 0
-        # Provide password, expect successful registration and subscription prompt
-        c2r.sendline(env.str("RHSM_PASSWORD"))
-        c2r.expect_exact("Enter number of the chosen subscription: ")
+
+        retries = 0
+        while True:
+            c2r.sendline(env.str("RHSM_USERNAME"))
+            print("Sending username:", env.str("RHSM_USERNAME"))
+            c2r.expect_exact("Password: ")
+            c2r.sendline()
+            try:
+                assert c2r.expect_exact(["Password", "Enter number of the chosen subscription"], timeout=300) == 0
+                # Provide password, expect successful registration and subscription prompt
+                c2r.sendline(env.str("RHSM_PASSWORD"))
+                print("Sending password")
+                c2r.expect_exact("Enter number of the chosen subscription: ")
+                break
+            except Exception:
+                retries = retries + 1
+                if retries == 3:
+                    raise
+                continue
```

Comment:
Does `raise` by itself reraise the exception or simply raise a new Exception with empty msg? Other than that the PR LGTM

Reply:
It will just re-raise any exception that was previously thrown in the code, if it has a message, then it will raise a message, otherwise, will just raise empty

Reply:
An naive test: 

```
[r0x0d@fedora p]$ python2 t.py 
raising with msg
Traceback (most recent call last):
  File "t.py", line 2, in <module>
    raise Exception("test")
Exception: test

[r0x0d@fedora p]$ python2 t.py 
no msg in raise
Traceback (most recent call last):
  File "t.py", line 8, in <module>
    raise Exception
Exception
```

And the test code: 

```python
[r0x0d@fedora p]$ cat t.py 
try:
    raise Exception("test")
except Exception:
    print("raising with msg")
    raise 

try:
    raise Exception
except Exception:
    print("no msg in raise")
    raise
```